### PR TITLE
Deprecate profile.name in favor of user.username

### LINKED ISSUE #13332: Real name (username) is used instead of displayname (nickname) for posts of users and under moderation /users
- [X] I have searched open and closed issues for duplicates

### Steps to Reproduce

1.  Switch do devlop branch
2.  do a 'git pull'

Actual Result:

The real names of the users are used as display name.
Both in Posts, and under / Moderation / Users.
As it seems, the field "username" and not "nickname" is pulled from the database.
It doesn't affect all users either, some still show the nickname under moderation / users.
I think it's because those users haven't looged in since the change.

i assume this was introduced somewhere around this merged pull request:

[#13328](https://github.com/friendica/friendica/commit/bf2c19b2e04031604c4a007f16fc7b2ba3c80137)

Expected Result:

Don´t show 'username' aka real name.

### Platform Info

Friendica Version:
2023-09.dev, latest commit
Friendica Source: from github
PHP version: 8.0 
SQL version: 10.5

### PR DESCRIPTION
**Pull request by MrPetovan**
Fix #13332 

This definitively puts to rest the confusion between `profile.name` and `user.username` by deprecating the former and making the latter the only source of update for the self-contact.

Besides the refactoring to get rid of yet another DI dependency in a module, there wasn't much to change, the `profile.name` field value itself was never read directly (instead using `contact.name` through the `owner` view), and was set only once in the Profile Settings module.

Unfortunately because of a difference in form field names in the user settings and the profile settings, Friendica users are expecting the `profile.name` field value to be used as their display name, so its value will be used to override the corresponding `user.username` value. This update is transparent as no change is done to the self-contact name, it's just the value in the user settings form field that will reflect the actual display name. From there on, any change to either form field will only update `user.username` and update the self-contact name accordingly.

### CODE REVIEW

File: update.php [hunk 1]
```diff
@@ -1349,3 +1349,20 @@ function update_1524(): int
 
 	return Update::SUCCESS;
 }
+
+function update_1525(): int
+{
+	if (!DBA::e('UPDATE `user` u
+    JOIN `profile` p
+    ON p.`uid` = u.`uid`
+    SET u.`username` = p.`name`')) {
```
Comment (annando) [inline update.php:1359]: I don't think that this is needed. For the owner view for example we are using the name from the self contact. And that is filled by the username. So I think that the username has the higher "priority" and shouldn't be overwritten by name in the profile.

**Review by annando — COMMENTED** [review]
I don't think that we should update the username. And when we do, we had to update the self contact as well, since it is feed out of the username. But we are already using the name from the self contact and the username and not the profile name any more. So for me it seems much safer not to perform this update.

**Comment by MrPetovan** [comment]
Let me try to explain the reason for this update again.

The situation in `2023.05` is the following:
- A form field called "Display Name" reads and writes `profile.name`.
- A form field called "Full Name" reads and writes `user.username`.
- The self-`contact.name` is used  as a display name.
- The self-`contact.name` is only updated from `profile.name`.

The situation in `develop` is exactly the same, but:
- The self-`contact.name` is only updated from `user.username`

This lead to some confusion from users that suddenly find the information they entered in the "Full Name" field as their actual display name rather than the information they entered in the "Display Name" field.

This means people using Friendica until version 2023.05 included expect the value in `profile.name` to be their display name. However, it makes more sense to use the `user.username` field from now on because there aren't multiple profiles with different display names anymore.

So in this PR:
- The profile "Display Name" form field now reads and writes "user.username".
- The user "Full Name" form field gets renamed to "Display Name" and still reads and writes "user.username".
- The self-`contact.name` is still used as a display name.
- The self-`contact.name` is still only updated from `user.username`.

But people still expect the value that was in `profile.name` to be their display name, so to avoid any confusion, the `user.username` value gets replaced by the `profile.name` value and the latter field is blanked out. This doesn't change any actual display name, neither for the people upgrading from `2023.05` nor for the people upgrading from `develop`, but it makes both "Display Name` form fields read the same value from the former single "Display Name" field.

**Comment by Raroun** [comment]
@annando 
This is really a kind of data protection desater.
I would be very happy if the pull request goes through so that the old display name reappears in the Ui for everyone.
I've had countless discussions as to why the real name was now used in public posts - without user consent.
There are also some users, which despite email from me, still have not noticed that they post under Real name.
Probably Apps like Tusky, Fedilab and so on don't give that - so you don't actually notice the change :)

**Comment by MrPetovan** [comment]
Fix https://squeet.me/display/962c3e10-2064-e0df-40bb-a7d491987948

**Comment by annando** [comment]
Nonetheless we have to update the name in the self contact as well, when we update the username from the profile name.

**Comment by MrPetovan** [comment]
For people on 2023.05, it's already the same, and for the people on `develop` who noticed the difference and have updated `user.username` to match `profile.name`, it will also be the same.

Anyone who notices a difference afterwards can change the display name from either of the two fields and it's going to be picked up by the self-contact.

**Comment by annando** [comment]
Would be great if you could extend the update to update the self contact as well.

**Comment by MrPetovan** [comment]
Done.

**Comment by annando** [comment]
Currently the database check fails. Seems like some missing updated documentation file.

**Comment by MrPetovan** [comment]
Should be fixed.

**Pull request merged**